### PR TITLE
Simple (SPEC-like) scans are more customizable

### PR DESCRIPTION
**Pull request by danielballan**
In current master, the subscriptions attached to the simple scans are hard-coded. Customizing them is tricky because they inspect the scan to automatically do things like use `motor` attribute as the x axis in `LivePlot`.

This introduces two new user-modifiable attributes: `.subs`, which works the same way as usual, and `.sub_factories`, which expects a *function that accepts a scan instance and returns a callback*. It's the job of the function to inspect the scan and pass the right arguments to, for example, `LivePlot`. A collection of these "sub factories" is included in the PR. See for example `plot_first_motor`.

I don't imagine that users will be writing very many "sub factories," but they are crucial to providing the magical scan inspection in a way that lets users add and remove other subs (like peak stats) at will.

### CODE REVIEW

**Comment by danielballan** [comment]
P.S. Tests pass. This is ready once it survives review.

**Comment by tacaswell** [comment]
bike shedding the name, `_BoundScan` might be better?

**Comment by klauer** [comment]
:+1: for sub-factories (I've been doing the same in a less-than-ideal way...)

File: bluesky/simple_scans.py [hunk 1]
```diff
@@ -47,12 +84,18 @@ def __call__(self, *args, subs=None, **kwargs):
...
 
-        self.scan = self._scan_class(gs.DETS, *args, **scan_kwargs)
-        if subs is None:
-            subs = self.subs
+        self.scan = self.scan_class(gs.DETS, *args, **scan_kwargs)
+        # Combine subs passed as args and subs set up in subs attribute.
+        _subs = normalize_subs_input(subs)
+        _subs.update(normalize_subs_input(self.subs))
+        # Create a sub from each sub_factory.
+        _subs.update({k: [sf(scan) for sf in v] for k, v in
```
Comment (klauer) [inline bluesky/simple_scans.py]: `scan` -> `self.scan`, this line and next

Comment (danielballan) [inline bluesky/simple_scans.py]: Good catch. Fixed and force-pushed.

**Pull request merged**